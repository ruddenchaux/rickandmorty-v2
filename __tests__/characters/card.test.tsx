import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import CharacterCard from '../../app/characters/card';

function encodeUri(str: string) {
  return encodeURI(str).replace(/\//g, '%2F').replace(/:/g, '%3A');
}

describe('CharacterCard', () => {
  it('should have the media', () => {
    const character = {
      id: '1',
      name: 'Rick',
      status: 'Alive',
      species: 'Human',
      image: 'https://rickandmortyapi.com/api/character/avatar/1.jpeg',
      location: { name: 'Earth' },
      episode: [{ name: 'Pilot' }],
    };

    render(<CharacterCard character={character} />);

    const image = screen.getByAltText('Rick');

    expect(image).toBeInTheDocument();
    expect(image).toBeInTheDocument();
    expect(image).toHaveAttribute(
      'src',
      expect.stringContaining(
        encodeUri('https://rickandmortyapi.com/api/character/avatar/1.jpeg')
      )
    );
  });

  it('should have the name', () => {
    const character = {
      id: '1',
      name: 'Rick',
      status: 'Alive',
      species: 'Human',
      image: 'https://rickandmortyapi.com/api/character/avatar/1.jpeg',
      location: { name: 'Earth' },
      episode: [{ name: 'Pilot' }],
    };

    render(<CharacterCard character={character} />);

    const name = screen.getByText('Rick');

    expect(name).toBeInTheDocument();
  });

  it('should have the status and species', () => {
    const character = {
      id: '1',
      name: 'Rick',
      status: 'Alive',
      species: 'Human',
      image: 'https://rickandmortyapi.com/api/character/avatar/1.jpeg',
      location: { name: 'Earth' },
      episode: [{ name: 'Pilot' }],
    };

    render(<CharacterCard character={character} />);

    const statusAndSpecies = screen.getByText('Alive - Human');

    expect(statusAndSpecies).toBeInTheDocument();
  });

  it.skip('should have the add to favorite action', () => {
    const character = {
      id: '1',
      name: 'Rick',
      status: 'Alive',
      species: 'Human',
      image: 'https://rickandmortyapi.com/api/character/avatar/1.jpeg',
      location: { name: 'Earth' },
      episode: [{ name: 'Pilot' }],
    };

    render(<CharacterCard character={character} />);

    const button = screen.getByRole('button', { name: 'Add to Favourite' });

    expect(button).toBeInTheDocument();
  });

  it('snapshot', () => {
    const character = {
      id: '1',
      name: 'Rick',
      status: 'Alive',
      species: 'Human',
      image: 'https://rickandmortyapi.com/api/character/avatar/1.jpeg',
      location: { name: 'Earth' },
      episode: [{ name: 'Pilot' }],
    };

    const { container } = render(<CharacterCard character={character} />);
    expect(container).toMatchSnapshot();
  });
});
