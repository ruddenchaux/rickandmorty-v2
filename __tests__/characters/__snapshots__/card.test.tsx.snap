// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterCard snapshot 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1fyw31w-MuiPaper-root-MuiCard-root"
  >
    <div
      class="MuiCardMedia-root css-1sr3gul-MuiCardMedia-root"
    >
      <img
        alt="Rick"
        data-nimg="1"
        decoding="async"
        height="280"
        loading="lazy"
        src="/_next/image?url=https%3A%2F%2Frickandmortyapi.com%2Fapi%2Fcharacter%2Favatar%2F1.jpeg&w=640&q=75"
        srcset="/_next/image?url=https%3A%2F%2Frickandmortyapi.com%2Fapi%2Fcharacter%2Favatar%2F1.jpeg&w=384&q=75 1x, /_next/image?url=https%3A%2F%2Frickandmortyapi.com%2Fapi%2Fcharacter%2Favatar%2F1.jpeg&w=640&q=75 2x"
        style="color: transparent;"
        width="280"
      />
    </div>
    <div
      class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
    >
      <h2
        class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
      >
        Rick
      </h2>
      <span
        class="MuiTypography-root MuiTypography-subtitle2 css-hav9kw-MuiTypography-root"
      >
        Alive
         - 
        Human
      </span>
    </div>
  </div>
</div>
`;
